packages/Search/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -205,15 +203,21 @@ KOKKOS_INLINE_FUNCTION int
 queryDispatch( BVH<DeviceType> const bvh, Predicate const &pred,
                Insert const &insert, SpatialPredicateTag )
 {
-    return spatial_query( bvh, pred, insert );
+    return spatialQuery( bvh, pred, insert );
 }
 
 template <typename DeviceType, typename Predicate, typename Insert>
 KOKKOS_INLINE_FUNCTION int
 queryDispatch( BVH<DeviceType> const bvh, Predicate const &pred,
                Insert const &insert, NearestPredicateTag )
 {
-    return nearestQuery( bvh, pred._query_point, pred._k, insert );
+    auto const geometry = pred._geometry;
+    auto const k = pred._k;
+    return nearestQuery( bvh,
+                         [geometry]( Node const *node ) {
```

Comment:
Does this lambda have to be a `KOKKOS_LAMBDA`?

Reply:
I don't think so since `queryDispatch()` is a `__host__ __device__` function.